pkg/virt-controller/leaderelectionconfig/readiness.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package leaderelectionconfig
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/emicklei/go-restful"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+)
+
+func ControllerLeadElectionReadinessFunc(_ *restful.Request, response *restful.Response) {
+	res := map[string]interface{}{}
+	cli, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		unhealthy(err, response)
+		return
+	}
+
+	endpoints, err := cli.CoreV1().Endpoints(DefaultNamespace).Get(DefaultEndpointName, metav1.GetOptions{})
```

Comment:
That's a nice idea. I think something more simple would be appropriate too, since the connectivity is already covered with the health check.

Maybe just switching a boolean to true after the leader election is won might already be good enough. What do you  think?